pkg/libovsdb/notation.go

```diff
@@ -101,12 +106,60 @@ func NewMutation(column string, mutator string, value interface{}) []interface{}
 	return []interface{}{column, mutator, value}
 }
 
+// Transact represents the request of a Transact call
+type Transact struct {
+	DBName     string
+	Operations []Operation
+}
+
+func NewTransact(params []interface{}) (*Transact, error) {
+	if len(params) < 2 {
+		return nil, errors.New("malformed transaction")
+	}
+
+	tx := new(Transact)
+	for i, v := range params {
+		switch i {
+		case 0:
+			dbname, ok := v.(string)
+			if !ok {
+				return nil, errors.New("malformed transaction")
+			}
+			tx.DBName = dbname
+		default:
+			b, err := json.Marshal(v)
+			if err != nil {
+				return nil, errors.New("malformed transaction")
+			}
+			var op Operation
+			err = json.Unmarshal(b, &op)
+			if err != nil {
+				return nil, errors.New("malformed transaction")
+			}
+			tx.Operations = append(tx.Operations, op)
+		}
+	}
+
+	return tx, nil
+}
+
 // TransactResponse represents the response to a Transact Operation
 type TransactResponse struct {
 	Result []OperationResult `json:"result"`
 	Error  string            `json:"error"`
```

Comment:
so how does one know that the operation failed ?

Reply:
From the previous spec.
... and a "result" member that is an array with the same number of elements as "params".  Each element of the "result" array corresponds to the same element of the "params" array.  The "result" array elements may be
   interpreted as follows:

   o  A JSON object that does not contain an "error" member indicates
      that the operation completed successfully.  The specific members
      of the object are specified below in the descriptions of
      individual operations.  Some operations do not produce any
      results, in which case the object will have no members.
   o  An <error> indicates that the matching operation completed with an
      error.
   o  A JSON null value indicates that the operation was not attempted
      because a prior operation failed.

Reply:
ok